sql/2020/03_Markup/pages_element_count_by_device_and_element_type_frequency.sql

```diff
@@ -0,0 +1,40 @@
+#standardSQL
+# Top used elements
+# See related: sql/2019/03_Markup/03_02b.sql
+
+CREATE TEMP FUNCTION AS_PERCENT (freq FLOAT64, total FLOAT64) RETURNS FLOAT64 AS (
+  ROUND(SAFE_DIVIDE(freq, total), 4)
+);
+
+CREATE TEMPORARY FUNCTION get_element_types_info(element_count_string STRING)
+RETURNS ARRAY<STRUCT<name STRING, freq INT64>> LANGUAGE js AS '''
+try {
+    if (!element_count_string) return []; // 2019 had a few cases
+
+    var element_count = JSON.parse(element_count_string); // should be an object with element type properties with values of how often they are present
+
+    if (Array.isArray(element_count) || typeof element_count != 'object') return [];
+
+    return Object.entries(element_count).map(([name, freq]) => ({name, freq})); 
+    
+} catch (e) {
+    return []; 
+}
+''';
+
+SELECT
+  _TABLE_SUFFIX AS client,
+  element_type_info.name,
+  SUM(element_type_info.freq) AS freq_m201, # total count from all pages
+  AS_PERCENT(SUM(element_type_info.freq), SUM(SUM(element_type_info.freq)) OVER (PARTITION BY _TABLE_SUFFIX)) AS pct_m202
```

Comment:
This will be biased towards elements that are used a lot, which may be what you're looking for. Instead, consider only counting elements once per page, so these stats would be in terms of the number/percent of pages having the element.

Reply:
pages_element_count_by_device_and_element_type_present.sql does that. They wanted it both ways.